index/secondary_index_manager.cc

```diff
@@ -68,6 +69,52 @@ void secondary_index_manager::add_index(const index_metadata& im) {
     _indices.emplace(im.name(), index{index_target_name, im});
 }
 
+static const data_type collection_keys_type(const abstract_type& t) {
```

Comment:
Most likely, won't build due to static function defined but not used.

Reply:
Seems reasonable, however 
```
~/c/scylla ((1a471907…)|✚1…) $ rg collection_keys_type
index/secondary_index_manager.cc
72:static const data_type collection_keys_type(const abstract_type& t) {
75:            throw std::logic_error(format("collection_keys_type: only collections (maps, lists and sets) supported, but received {}", t.cql3_type_name()));
~/c/scylla ((1a471907…)|✚1…) $ ninja -vvv build/release/index/secondary_index_manager.o && echo success
[1/2] /usr/bin/ninja -C build/release/seastar seastar
ninja: Entering directory `build/release/seastar'
ninja: no work to do.
[2/2] clang++ -MD -MT build/release/index/secondary_index_manager.o -MF build/release/index/secondary_index_manager.o.d -I/home/enedil/code/scylla/seastar/include -I/home/enedil/code/scylla/build/release/seastar/gen/include -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -Werror=unused-result -DSEASTAR_API_LEVEL=6 -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DFMT_LOCALE -DFMT_SHARED -I/usr/include/p11-kit-1 -Wl,--gc-sections -ffunction-sections -fdata-sections  -O3 -mllvm -inline-threshold=225 -fno-slp-vectorize -DSCYLLA_ENABLE_WASMTIME -g -gz -iquote. -iquote build/release/gen --std=gnu++20 -ggdb3 -fno-omit-frame-pointer -ffile-prefix-map=/home/enedil/code/scylla=.  -march=westmere -DBOOST_TEST_DYN_LINK   -Iabseil -fvisibility=hidden  -Wall -Werror -Wno-mismatched-tags -Wno-tautological-compare -Wno-parentheses-equality -Wno-c++11-narrowing -Wno-sometimes-uninitialized -Wno-return-stack-address -Wno-missing-braces -Wno-unused-lambda-capture -Wno-overflow -Wno-noexcept-type -Wno-error=cpp -Wno-ignored-attributes -Wno-overloaded-virtual -Wno-unused-command-line-argument -Wno-defaulted-function-deleted -Wno-redeclared-class-member -Wno-unsupported-friend -Wno-unused-variable -Wno-delete-non-abstract-non-virtual-dtor -Wno-braced-scalar-init -Wno-implicit-int-float-conversion -Wno-delete-abstract-non-virtual-dtor -Wno-uninitialized-const-reference -Wno-psabi -Wno-narrowing -Wno-array-bounds -Wno-nonnull -Wno-uninitialized -Wno-error=deprecated-declarations -DXXH_PRIVATE_API -DSEASTAR_TESTING_MAIN -DHAVE_LZ4_COMPRESS_DEFAULT  -c -o build/release/index/secondary_index_manager.o index/secondary_index_manager.cc
success
```

Reply:
Maybe clang is more lenient than gcc here.

Or maybe this got leaked into our headers:

```
$ git grep unused-function origin/master
origin/master:alternator/expressions.g:    #pragma GCC diagnostic ignored "-Wunused-function"
origin/master:cql3/Cql.g:#pragma GCC diagnostic ignored "-Wunused-function"
```

Reply:
```console
$ grep -r unused-function build/dev/gen
build/dev/gen/cql3/Cql.g:#pragma GCC diagnostic ignored "-Wunused-function"
build/dev/gen/cql3/CqlLexer.cpp:#pragma GCC diagnostic ignored "-Wunused-function"
build/dev/gen/cql3/CqlLexer.hpp:#pragma GCC diagnostic ignored "-Wunused-function"
build/dev/gen/alternator/expressions.g:    #pragma GCC diagnostic ignored "-Wunused-function"
build/dev/gen/alternator/expressionsLexer.cpp:    #pragma GCC diagnostic ignored "-Wunused-function"
build/dev/gen/alternator/expressionsLexer.hpp:    #pragma GCC diagnostic ignored "-Wunused-function"
```